### PR TITLE
[pytorch][build] DLC for PT 1.8.2 training

### PR DESCRIPTION
**Pull request by waytrue17**
Torchvision 0.9.2 (compatible with torch 1.8.2) has not been released yet. Currently using torchvision built from v0.9.2-rc2 in this PR

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by waytrue17** [comment]
Can you please review @saimidu

**Review by SergTogul — APPROVED** [review]

File: pytorch/buildspec.yml [hunk 1]
```diff
@@ -71,40 +71,40 @@ images:
...
-    device_type: &DEVICE_TYPE gpu
-    python_version: &DOCKER_PYTHON_VERSION py3
-    tag_python_version: &TAG_PYTHON_VERSION py38
-    cuda_version: &CUDA_VERSION cu111
-    os_version: &OS_VERSION ubuntu20.04
-    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION ]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile., *DEVICE_TYPE ]
-    context:
-      <<: *INFERENCE_CONTEXT
+  # BuildCPUPTInferencePy3DockerImage:
```
Comment (tejaschumbalkar) [inline pytorch/buildspec.yml:86]: Please revert the commented code block

File: pytorch/buildspec.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK pytorch
-version: &VERSION 1.9.0
-short_version: &SHORT_VERSION 1.9
+version: &VERSION 1.8.2
```
Comment (tejaschumbalkar) [inline pytorch/buildspec.yml:4]: Create a new file `buildspec-1-8-2.yml` to store the v1.8.2 configuration and revert the changes in `pytorch/buildspec.yml` to reflect the latest PT version v1.9.0

Comment (waytrue17) [reply pytorch/buildspec.yml]: Done

File: pytorch/buildspec.yml (hunk 1 above)
Comment (waytrue17) [reply pytorch/buildspec.yml]: Done

**Comment by waytrue17** [comment]
> Can we include the fix for broken SM utility packages #1307 in this PR?

Added the numba version in docker files

File: pytorch/buildspec-1-9-0.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
+account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (shantanutrip) [inline pytorch/buildspec-1-9-0.yml:1]: Is this an exact replica of `buildpec.yml` ? Is it required @tejaschumbalkar ?

**Comment by tejaschumbalkar** [comment]
PT 1.8 is out of support
